types/factory.go

```diff
@@ -5,12 +5,20 @@ import (
 	"reflect"
 )
 
+var enableCache = false
+
+// ToggleCache enables caching of ssz hash tree root. It is disabled by default.
+func ToggleCache(val bool) {
```

Comment:
this is only for testing ,correct ?

Reply:
Hmm to some extent -  it is nice to have the functionality of disabling caching though, either for benchmark reasons or for other reasons. It is the best compromise I found to prevent collision resistance in tests and maintain speed.

Reply:
sounds good. Would be an easy way to debug the cache if we have anything wrong